### DISCUSSION TITLE
Exult 1.8.0 stuck on creating AudioMixer

### DISCUSSION THREAD

**Discussion by yeldarbthebarbarian** [exult/exult#338] [General]
Just compiled the most recent version of Exult. but cannot for the life of me get it to launch, it just hangs here. I am running Ubuntu 22.04.2 LTS
here is the log. 

Exult version 1.8.0
Built at: Apr  6 2023 20:30:40
Compile-time options: USE_TIMIDITY_MIDI, USE_FMOPL_MIDI, HAVE_ZIP_SUPPORT
Compiler: GCC 11.3.0

Platform: Linux version 5.19.0-32-generic 
Exult path settings:
Data          : /usr/local/share/exult
Digital music : /usr/local/share/exult/music

Looking for 'blackgate' at '/usr/local/share/exult/blackgate'... found game with identity 'FORGE'
Looking for 'forgeofvirtue' at '/usr/local/share/exult/forgeofvirtue'... but it wasn't there.
Could not open file '/usr/local/share/exult/forgeofvirtue/static/mainshp.flx
Looking for 'serpentisle' at '/usr/local/share/exult/serpentisle'... but it wasn't there.
Could not open file '/usr/local/share/exult/serpentisle/static/mainshp.flx
Looking for 'silverseed' at '/usr/local/share/exult/silverseed'... but it wasn't there.
Could not open file '/usr/local/share/exult/silverseed/static/mainshp.flx
Looking for 'serpentbeta' at '/usr/local/share/exult/serpentbeta'... but it wasn't there.
Could not open file '/usr/local/share/exult/serpentbeta/static/mainshp.flx
Black Gate   : not found (<BLACKGATE_STATIC>/)
Forge of Virtue   : found
exult_bg.flx : found
Serpent Isle   : not found (<SERPENTISLE_STATIC>/)
Silver Seed   : not found (<SILVERSEED_STATIC>/)
Checking rendering support
 Windowed	
SDL Reports 640x400 16 bpp and 32 bpp windowed surfaces are not OK. Windowed scalers may not work properly.
SDL Reports no usable fullscreen resolutions.
SDL Reports 1024x768 16 bpp and 32 bpp windowed surfaces are not OK. Attempting to use 16 bpp. anyway
Creating AudioMixer...

I would also like to add it may be completely user error and I using this as an opportunity to learn more about compiling programs as I am a newbie to it. this will also be my first playthrough of Ultima 7

**Comment by Dragon-Baroque** [comment]
It look like SDL2 - the Audio and Graphics layer - neither finds a suitable screen resolution, nor finds a suitable audio device.

Could you post the `./configure` command you used and the few report lines that `./configure` produces at its end, like this :
```
Exult v1.8.0

SDL........................ : 2.26.3
Ogg.........................: 1.3.5
Vorbis......................: 1.3.7
...

Now type 'make' to build Exult
```
The reason is : your `Compile-time options: USE_TIMIDITY_MIDI, USE_FMOPL_MIDI, HAVE_ZIP_SUPPORT` is very short, I expect at least `USE_ALSA_MIDI` in it. Either your `./configure` has an explicit `--disable-alsa` in it, or it is a side effect of _your build station lacking_ the `-dev` packages.

In `.github/workflows/ci-linux.yml`, you have a good full build that you could use as a basis, both for ensuring that the required packages are there ( I have removed below the requirements for the Exult Studio + the Gimp Plugin, but you could take it in full, that would be a good check ) :
```
sudo apt-fast install -y \
zlib1g-dev libogg-dev libvorbis-dev libasound2-dev libfluidsynth-dev libsdl2-dev libpng-dev \
libfreetype6-dev libxml2-dev bison flex timidity
```
and for building Exult with maximum debug ( I have removed the build of Exult Studio + the Gimp Plugin, but again you could try to build Exult and Studio )
```
./configure --with-debug=extreme --enable-compiler --enable-zip-support  \
--enable-shared --enable-midi-sfx --enable-data --enable-mods --enable-usecode-container \
--enable-nonreadied-objects --disable-oggtest --disable-vorbistest
```
The crux here is `--with-debug=extreme` for more startup / run messages at least until your Exult starts...

Next, your run does not list any supported graphic resolution, mine does :
```
Checking rendering support
 Windowed
 640x480         8 bpp ok 16 bpp ok 32 bpp ok
 720x480         8 bpp ok 16 bpp ok 32 bpp ok
 ...
 1920x1080       8 bpp ok 16 bpp ok 32 bpp ok
SDL Reports 640x400 16 bpp and 32 bpp windowed surfaces are not OK. Windowed scalers may not work properly.
Creating AudioMixer...
Audio opened using format: 44100 Hz 2 Channels
```
Would you by any chance be using a portrait style screen ( higher that it is wide ) ? That could explain this.

Finally, I do not want to install Exult in the ( default ) `/usr/local` which is protected. I install it into some directory of my `$HOME`, using `./configure --prefix=<the full path to the target installation directory>`, so that I do not need to run `make install` as superuser.
Be warned though, that doing so, and building in full, the `make install` shall fail on the Gimp Plugin + the Gnome Thumbnailer installations.
